.github/workflows/demo.yml

```diff
@@ -0,0 +1,123 @@
+name: Pulp Python 101
+
+on:
+  schedule:
+    - cron: '30 2 * * 0'
+
+env:
+  COLORTERM: 'yes'
+  PY_COLORS: '1'
+  TERM: 'xterm-256color'
+
+jobs:
+  demo:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+
+      - name: Starting Pulp server
+        run: |
+            mkdir settings pulp_storage pgsql containers
+            echo "CONTENT_ORIGIN='https://pulp.example.com'" >> settings/settings.py
+            docker run --detach \
+                      --name pulp \
+                      --publish 443:443 \
+                      --volume "/$(pwd)/settings:/etc/pulp:Z" \
+                      --volume "/$(pwd)/pulp_storage":/var/lib/pulp:Z \
+                      --volume "/$(pwd)/pgsql":/var/lib/pgsql:Z \
+                      --volume "/$(pwd)/containers":/var/lib/containers:Z \
+                      --device /dev/fuse \
+                      pulp/pulp:https
+
+            # Wait server to be up
+            sleep 10
+            for _ in $(seq 20)
+            do
+              sleep 2
+              if curl --insecure --fail https://localhost/pulp/api/v3/status/ > /dev/null 2>&1
```

Comment:
This is the step where we set the Pulp server.
The other steps are client steps